src/Components/Blueprints/BlueprintVersionFilter.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownList,
+  MenuToggle,
+  MenuToggleElement,
+} from '@patternfly/react-core';
+
+interface blueprintVersionFilterProps {
+  blueprintVersionFilter?: string | undefined;
+  setBlueprintVersionFilter?: React.Dispatch<
+    React.SetStateAction<string | undefined>
+  >;
+}
+
+export const ALL_VERSIONS_CONST = 'All Versions';
+export const NEWEST_VERSION_CONST = 'Newest';
+const LATEST_VERSION = -1;
+
+export const getBlueprintVersionFilterAPIValue = (
+  blueprintVersionFilter: string | undefined
+): number | undefined => {
+  if (
+    !blueprintVersionFilter ||
+    blueprintVersionFilter === ALL_VERSIONS_CONST
+  ) {
+    return undefined;
+  } else if (blueprintVersionFilter === NEWEST_VERSION_CONST) {
+    return LATEST_VERSION;
+  }
+  return parseInt(blueprintVersionFilter);
+};
+
+const BlueprintVersionFilter: React.FC<blueprintVersionFilterProps> = ({
+  blueprintVersionFilter,
+  setBlueprintVersionFilter,
+}: blueprintVersionFilterProps) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const onSelect = (
+    _event: React.MouseEvent<Element, MouseEvent> | undefined,
+    value: string | undefined
+  ) => {
+    if (setBlueprintVersionFilter) setBlueprintVersionFilter(value);
+    setIsOpen(false);
+  };
+
+  return (
+    <div data-testid="blueprint-version-filter">
```

Comment:
No need for a `<div>` here (we never use those), you can just put the `data-testid` directly into the `<Dropdown>`. You might need to surround the component with an empty set of `<></>` tags.

Reply:
yup `Unable to find an element by: [data-testid="blueprint-version-filter"]` when I move the testid

Reply:
Urgh, that's frustrating. Do you actually need a data-testid? I usually use the Testing Playground browser plugin (https://chromewebstore.google.com/detail/testing-playground/hejbmebodbijjdhflfknehhcgaklhano?pli=1) and try to avoid data-testid as much as possible... (the Testing Library folks consider using it to be a last resort: https://testing-library.com/docs/queries/bytestid/)

Reply:
Looks like a cool plugin, let me try! :)